### PR TITLE
Preapps 503 Edit appointments

### PR DESCRIPTION
**Pull request by silentsakky**
- Open appointment edit view on double click
- add context menu for appointments with edit option
- Add edit button on appointment details tooltip
- create graphql decorators for create/update/delete appointment
- modify graphql schema to add alarm data as part of InviteComponent
- add functionality to create exception in recurring series
- add support for editing of appointments
- add support for showing exceptions in calendar

### CODE REVIEW

File: src/normalize/entities.ts [hunk 1]
```diff
@@ -54,8 +50,7 @@ const RecurrenceInfo = new Entity({
 });
 
 const CalendarItemAlarm = new Entity({
-	trigger: CalendarItemAlarmTrigger,
-	at: ['attendees', CalendarItemAlarmAttendees]
```
Comment (pl12133) [inline src/normalize/entities.ts]: Any reason to remove this? It seems like a valid property on an `<alarm>`.

Comment (billneff79) [reply src/normalize/entities.ts]: - [ ] let's put this back.  It's in the graphql schema for the output

File: src/schema/schema.graphql [hunk 2]
```diff
@@ -201,8 +201,27 @@ type Instance {
 	start: Float
 	dueDate: Float
 	tzoDue: Int
-	isRecurring: Boolean
```
Comment (billneff79) [inline src/schema/schema.graphql]: Either this needs to stay in, or we should stop denormalizing it in the entity definition above

File: src/schema/schema.graphql [hunk 3]
```diff
@@ -864,6 +886,12 @@ input CalendarItemDateTimeInput {
 	date: String!
 }
 
+# @FIXME
```
Comment (billneff79) [inline src/schema/schema.graphql]: - [ ] Remove this comment

File: src/schema/schema.graphql (hunk 2 above)
Comment (silentsakky) [reply src/schema/schema.graphql]: Removed from normalisation logic

File: src/schema/generated-schema-types.ts [hunk 4]
```diff
@@ -480,8 +498,26 @@ export interface Instance {
 	start?: number | null;
 	dueDate?: number | null;
 	tzoDue?: number | null;
-	isRecurring?: boolean | null;
 	utcRecurrenceId?: string | null;
+	alarm?: boolean | null;
```
Comment (billneff79) [inline src/schema/generated-schema-types.ts]: Do we really need all of these ` | null`s?  What is that accomplishing for us?

Comment (billneff79) [reply src/schema/generated-schema-types.ts]: nevermind - this is the generated file :)

**Pull request merged**